tests/typechecker/none_coalescing_assign_to_immutable.err

```diff
@@ -0,0 +1,7 @@
+Error: Assignment between incompatible types (‘String?’ and ‘String’)
```

Comment:
Seems to me that the error here should be that you're trying to modify an immutable variable, not that the types don't match. (Because the types *do* match!)

This isn't because of your changes though as far as I can tell, just an observation.